### PR TITLE
Refactor main to prepare for new command

### PR DESCRIPTION
**Pull request by glrf**
## Checklist

- [ ] Keep pull requests small so they can be easily reviewed.
- [ ] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [ ] Update the documentation.
- [ ] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -241,114 +46,112 @@ func main() {
...
+	flag.Parse()
+	if flag.NArg() == 0 {
 		flag.Usage()
 		os.Exit(2)
 	}
 
-	configFile := flag.Arg(0)
+	setupLogger()
 
+	WaitForKeepalivedTermination(ctx, stop)
```
Comment (simu) [inline main.go:83]: I'm not 100% sure that this needs to be in `main()`, but it probably doesn't hurt.

File: notification.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	Status   NotificationStatus
+}
+
+type NotificationStatus string
+
+const (
+	NotificationMaster NotificationStatus = "MASTER"
+	NotificationFault                     = "FAULT"
+	NotificationBackup                    = "BACKUP"
+)
```
Comment (simu) [inline notification.go:24]: I wanted to double-check that this works how I thought it works (i.e. all consts get type `NotificationStatus`), but with my simple test program which does `fmt.Printf("%T %T %T", NotificationMaster, NotificationFault, NotificationBackup)` I get `main.NotificationStatus string string`, and not the expected 3x `main.NotificationStatus`. Notably, a lot of documentation online still uses the form you've used here, so I'm now really unsure whether we should have explicit type `NotificationStatus` for all consts here.

**Review by simu — APPROVED** [review]
LGTM overall

File: notification.go (hunk 2 above)
Comment (glrf) [reply notification.go:24]: You're right. These should also be explicitly typed. I think I made that mistake before.  I somehow though the `iota` syntactic sugar applied here too, which isn't the case

Without it they are untyped constant, which means they can be used as both a `string` and a `NotificationStatus`, so it worked in all cases here, but it's better to be explicit here.

File: main.go (hunk 1 above)
Comment (glrf) [reply main.go:83]: Yeah not entirely happy with this either, but I don't really want to pass the cancel function further down the stack

File: e2e_test.go [hunk 3]
```diff
@@ -100,13 +100,19 @@ func runCmd(cmd *exec.Cmd) (*bytes.Buffer, func() error, error) {
...
 	for count < n {
 		var line string
 		var err error
 		line, err = buf.ReadString('\n')
 		for errors.Is(err, io.EOF) {
 			line, err = buf.ReadString('\n')
+			if ctx.Err() != nil {
+				assert.NoError(t, ctx.Err())
+				return
+			}
```
[the remark this answers is not in the document]
Comment (bastjan) [reply e2e_test.go:115]: Both not super readable 🙈

**Review by bastjan — APPROVED** [review]

**Pull request merged**